### PR TITLE
NOISSUE-1 Fixed behavior

### PR DESCRIPTION
**Pull request by schuellerf**
Several fixes to the behavior, removed SMELL and updated wording

please comment if these an eligible changes

### CODE REVIEW

**Comment by schuellerf** [comment]
... also the "bad" commit messages don't cause an "exit 1" but that might be on purpose?

**Comment by ochosi** [comment]
> ... also the "bad" commit messages don't cause an "exit 1" but that might be on purpose?

The idea was to not really block on Jira tickets to not make life hard/er for external contributors. They _can_ create Jira issues (because our project is public), but do we really want to force them to?

File: .github/workflows/best-practice.yml [hunk 1]
```diff
@@ -13,23 +13,19 @@ jobs:
         with:
           ref: ${{ github.event.pull_request.head.sha }}
           fetch-depth: 0
-      - run: | # SMELL: this is a bit stupid, but apparently git rev-list needs it?
-          git branch -a
-          git checkout main
-          git checkout readme
       - name: PR title contains Jira ticket
-        run: if [ -z $(grep -Eo "[A-Z][A-Z0-9]+-[0-9]+" <<< "${{ github.event.pull_request.title }}") ]; then echo "PR title should contain a Jira."; else echo "Well done!"; fi
+        run: if [ -z $(grep -Eo "[A-Z][A-Z0-9]+-[0-9]+" <<< "${{ github.event.pull_request.title }}") ]; then echo "Please add a Jira ticket to the PR title."; exit 1;else echo "Well done!"; fi
```
Comment (ochosi) [inline .github/workflows/best-practice.yml]:
```suggestion
        run: if [ -z $(grep -Eo "[A-Z][A-Z0-9]+-[0-9]+" <<< "${{ github.event.pull_request.title }}") ]; then echo "Please add a Jira ticket to the PR title."; else echo "Well done!"; fi
```

**Review by ochosi — CHANGES_REQUESTED** [review]

**Review by ochosi — APPROVED** [review]
LGTM!

**Pull request merged**